sdks/java/extensions/sql/src/test/java/org/apache/beam/sdk/extensions/sql/meta/provider/pubsub/PubsubJsonIT.java

```diff
@@ -264,7 +273,7 @@ public void testSQLLimit() throws Exception {
                 });
 
     // wait one minute to allow subscription creation.
-    Thread.sleep(60 * 1000);
+    eventsTopic.checkIfAnySubscriptionExists(project, Duration.standardMinutes(1));
```

Comment:
It would be more correct to pull `project` out of the pipeline you're using here, rather then from `TestPipeline` globally. I believe `pipeline.getOptions().as(GcpOptions.class).getProject()` should work.

Reply:
The `project` option is set for TestPipelineOption from build.gradle so it's supposed to have the same value.

But yes, I agree that I should get `project` from existing `pipeline`.